hbase-balancer/src/test/java/org/apache/hadoop/hbase/master/balancer/TestStochasticLoadBalancerHeterogeneousCost.java

```diff
@@ -306,6 +245,10 @@ static class StochasticLoadTestBalancer extends StochasticLoadBalancer {
     private FairRandomCandidateGenerator fairRandomCandidateGenerator =
       new FairRandomCandidateGenerator();
 
+    StochasticLoadTestBalancer() {
+      super(new DummyMetricsStochasticBalancer());
+    }
```

Comment:
The balancer test suite actually couldn't be run in its entirety without this addition, because otherwise there are JMX conflicts. So this is just an unrelated bug fix that I forgot to separate in my planning

Reply:
Do we see the impact of this bug in our CI runs?

Reply:
Surprisingly we don't. I don't understand exactly how the CI env differs from my local env to make that the case, but presumably it has better isolation between the tests so that there's no conflict in the JMX setup when running many tests at once

Reply:
Maybe a difference between running in IDE vs directly in maven cli?